### ISSUE TITLE
TouchElements and another object combination is not work on Linux from 1.7.1.1(1.7.1.0)

### ISSUE THREAD

**Issue by ginga81** [leezer3/OpenBVE#435]
[TouchElements-testtrain.zip](https://github.com/leezer3/OpenBVE/files/4244633/TouchElements-testtrain.zip)
To the 1.7.0.3, this sample is works perfectly(without the Lineargauge's Behavior).

This sample of Panel.xml is using all objects.
At this sample, if remove the all objects outside of <Screen> section, Touchelements is works.
If the another objects, such as Pilotlamp put outside of Screen section, TouchElements is not work.
From @S520 says that at Windows, this issue is not happen.

Each object and results:
\<Screen\> section only: works well.
If the objects put at the inside of \<Screen\> section, DigitalGauge and Digitalnumber is same result as bellow.

The another objects are also works well at the inside of <Screen> section.

If the \<Screen\> section outside puts each elements

Pilotlamp: do not work Touchelements

Needle: works well

DigitalNumber:
Normal until an action occurs. Once the DigitalNumber condition is met, touch is no longer possible.
For example, if Digitalnumber's \<Subject\> is Sec, TouchElements can touch one second.
If Digitalnumber's \<Subject\> is SecD1, TouchElements can touch 10 seconds.
If the Digitalnumber is within the \<Screen\> section, if written screen has't select, the happening counter is not work, so that this truble is not happen.
If the Digitalnumber's written \<Screen\> section is selected, and event counter has been occured, the Touchelements is no longer possible.

DigitalGauge: do not work Touchelements

Timetable: works well

LinearGauge: works well

Sorry to trouble you, please check it.

**Comment by leezer3** [comment]
It's no problem.

I haven't had the chance to properly look at this yet, but please try the current build- This fixes a somewhat significant issue with the projection matrix in the cab, which could well be related.
https://github.com/leezer3/OpenBVE/commit/500636acb5e29d7183b75666796fc230b75eb5c9

I'd also suspect it's probably not Linux related per-se, much more likely to be being triggered by a rounding error with your screen resolution.

Anyway, will try and see if I can reproduce tomorrow for you.

**Comment by ginga81** [comment]
I tested the OpenBVE-2020-03-01.zip.
But, not works.
1.7.0.3 is works well, so I was going to check the old daily build.
But, the oldest daily build is 2019.10.1.
2019.10.1's version is not works well.
Luckily, I have downloaded OpenBVE-2019-09-22.zip at the past and I could test this.
At this build, the sample can works!
So I doubt the from 2019.9.22 to 2019.10.01.

**Comment by ginga81** [comment]
I tested same sample at 1.7.1.4.
but not works.
I cannot write toucherements tutorial...

**Comment by leezer3** [comment]
I've just tried this on Linux, with the most recent changes and all seems OK.

I then tried the following train in Panel2 extended mode:
https://sites.google.com/a/bve4trains.com/bve4trains/Home/class-47/cl47%20786%20West%20Coast%20Railway%20Co%20Light%20Engine%20OPEN.zip?attredirects=0&d=1

This also seems to work OK.

Can you please test both with the most recent build (after the Z-normal fix) and let me know what you get with both?

I'm hoping that this may have sorted the issue.
If not, hopefully it'll give me a little more insight into whether it's a wider issue, or something when the file gets loaded on your end.

**Comment by ginga81** [comment]
I am sorry.
I tested 1.7.1.4, one time I can very nice working.
At the later, I tested 1.7.1.4 again, cannot touch.
I am very confusing why cannot touch.
So I posted.
Today, I test 1.7.1.4 again, I found why this trouble happens.
If NewRenderer is check, touch is working.
If the NewRenderer is off, cannnot touch.
And I tested at OpenBVE-2020-04-26.zip, if the NewRenderer option is off, cannot touch.
It seems the Panel2 extend mode is not affect at this trouble.
